examples/gno.land/r/gnoland/faucet/faucet.gno

```diff
@@ -34,13 +35,11 @@ func Transfer(to std.Address, send int64) string {
 
 	var sendCoins std.Coins
 
-	if send <= limit.AmountOf("ugnot") {
-
+	// limit the pre request
```

Comment:
What do you think about simply sending the maximum limit when the requested amount is above the threshold, instead of denying the request?

You could simplify this code segment this way:

```go
// You can make this a separate helper method
getUgnotCoins := func(amount int64) std.Coins {
    return std.Coins { std.Coin { Denom: "ugnot", Amount: amount } }
}

var (
    maxAmount = GetPerTransferLimit()
    sendCoins = getUgnotCoins(maxAmount)
)

// Check if the requested amount 
 // is lower than the threshold
if send < maxAmount {
    sendCoins = getUgnotCoins(send)
}

// ...

```

Reply:
This is a good idea.  the max per account limit is set in the bot code.  per request, the limit is to guard someone who calls it from other clients other than the discord bot.  that could be easily abused if we give it to max amount here.  Also, getUgnotCoins is redundant code. The limit variable is a Coins type; it has a method called AmountOf().

Reply:
An other reason for bot checking the account limit off chain instead of sending request tx to the contract to verify account limit, I assume there will be abusive behaviors. Request gas and fees are paid by bot admin account, not by users.